packages/mui-styles/src/StylesProvider/StylesProvider.js

```diff
@@ -82,11 +80,9 @@ export default function StylesProvider(props) {
     }
 
     if (!context.jss.options.insertionPoint && injectFirst && typeof window !== 'undefined') {
-      if (!injectFirstNode) {
-        const head = document.head;
-        injectFirstNode = document.createComment('mui-inject-first');
-        head.insertBefore(injectFirstNode, head.firstChild);
-      }
+      const head = document.head;
+      const injectFirstNode = document.createComment('mui-inject-first');
+      head.insertBefore(injectFirstNode, head.firstChild);
```

Comment:
The original code was checking that the injection happened exactly once. After the change, the injection always runs. We need to revert this change. I think we're better off with adding an ESLint disable comment. See https://github.com/mui/material-ui/blob/bdf2fd4cf2be6cfc4769b2e2f98b2ba0c9ebce18/docs/data/material/components/use-media-query/UseWidth.tsx#L23-L24

Reply:
Thank you for the review and insights, I'll revert this and add the comments as requested.

Reply:
Hey @jlewins, this needs to be reverted.

Reply:
Both changes have been actioned, thank you for your help!